hw/ip/otbn/util/shared/control_flow.py

```diff
@@ -0,0 +1,380 @@
+#!/usr/bin/env python3
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import Dict, List
+
+from .decode import OTBNProgram, decode_elf
+from .insn_yaml import Insn
+from .section import CodeSection
+
+
+class ControlLoc:
+    '''Base class for control-flow locations.
+
+    Represents a normal PC, as a branch or jump instruction could link to.
+    '''
+    def __init__(self, pc: int):
+        self.pc = pc
+
+    def is_special(self):
+        return False
+
+    def pretty(self) -> str:
+        return '{:#x}'.format(self.pc)
```

Comment:
Is the min python version new enough to use fstrings? It probably is... or it should be at any rate.

Reply:
I can't find the min version exactly, but python 3 is required and looking at the Python docs I don't think anything here was added in a minor version, so I think we're safe:
https://docs.python.org/3/library/string.html#format-specification-mini-language